src/app/page.js

```diff
@@ -354,8 +354,18 @@ export default function Home() {
   }, []);
 
   const formattedReadData = readData.map((item) => {
-    const dataString = Array.from(item.data).map((byte) => byte.toString(16).padStart(2, '0').toUpperCase());
-    return '0x' + item.address.toString(16).padStart(4, '0').toUpperCase() + ': ' + dataString.join(' ');
+      const dataString = Array.from(item.data).map((byte) => byte.toString(16).padStart(2, '0').toUpperCase());
+
+      const a = (val) => (val & 0xFF).toString(16).padStart(2, "0");
+      const sum = (items) => Array.from(items).reduce((a, b) => a + b, 0);
+      let address = a(item.address >> 8) + a(item.address & 0xFF);
+      const crc = sum([item.data.length, (item.address >> 8) & 0xFF, (item.address & 0xFF), 0x00]) + sum(item.data);
+
+      return ':' + a(item.data.length)
+	  + address
+          + "85" /* FIXME: whats valid here?!? */
```

Comment:
The record type should always be `00` in our case - a [data record](https://en.wikipedia.org/wiki/Intel_HEX#Record_types).